### PR TITLE
CCIT-90: Add module template for platform v3

### PR DESCRIPTION
**Pull request by mvktsk**
Added Platform V3 new module PowerShell template.

### CODE REVIEW

File: {RepositoryName}/src/{Namespace}.Data/Repositories/MyVC3Module4DbContext.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+using EntityFrameworkCore.Triggers;
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Data/Repositories/MyVC3Module4DbContext.cs]: is the filename "MyVC3Module4DbContext.cs" correct?

File: {RepositoryName}/src/{Namespace}.Data/{Namespace}.Data.csproj [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Data/{Namespace}.Data.csproj:1]: remove "/src/{Namespace}.Data/packages.config" file

File: {RepositoryName}/src/{Namespace}.Web/Module.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            // register settings
+            var settingsRegistrar = appBuilder.ApplicationServices.GetRequiredService<ISettingsRegistrar>();
+            settingsRegistrar.RegisterSettings(ModuleConstants.Settings.AllSettings, ModuleInfo.Id);
+
+            // register permissions
+            var permissionsProvider = appBuilder.ApplicationServices.GetRequiredService<IPermissionsRegistrar>();
+            permissionsProvider.RegisterPermissions(ModuleConstants.Security.Permissions.AllPermissions.Select(x =>
+                new Permission()
+                {
+                    GroupName = "{ModuleId}.Web",
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Web/Module.cs]: redundant ".Web"?

File: {RepositoryName}/src/{Namespace}.Web/package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Web/package.json:1]: remove "/src/{Namespace}.Web/package-lock.json" file

File: {RepositoryName}/src/{Namespace}.Core/ModuleConstants.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                    Name = "{ModuleId}.{ModuleId}Enabled",
+                    GroupName = "Store|General",
+                    ValueType = SettingValueType.Boolean,
+                    DefaultValue = false
+                };
+
+                public static SettingDescriptor {ModuleId}Password = new SettingDescriptor
+                {
+                    Name = "{ModuleId}.{ModuleId}Password",
+                    GroupName = "Store|Advanced",
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Core/ModuleConstants.cs]: GroupName should include a variable

File: {RepositoryName}/src/{Namespace}.Web/module.manifest [hunk 6]
```diff
@@ -21,14 +21,6 @@
     <assemblyFile>{Namespace}.Web.dll</assemblyFile>
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Web/module.manifest:21]: VirtoCommerce.Core and VirtoCommerce.Platform.Core are not the same. Comment out 
`<dependency id="VirtoCommerce.Core" version="{CoreModuleVersion}" />`

**Review by eMazeika — CHANGES_REQUESTED** [review]
module name doesn't support "."
Data/Caching/ should include {module}CacheRegion.cs file

File: {RepositoryName}/src/{Namespace}.Data/Repositories/MyVC3Module4DbContext.cs (hunk 1 above)
Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Data/Repositories/MyVC3Module4DbContext.cs]: Fixed

File: {RepositoryName}/src/{Namespace}.Data/{Namespace}.Data.csproj (hunk 2 above)
Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Data/{Namespace}.Data.csproj:1]: Fixed

File: {RepositoryName}/src/{Namespace}.Web/Module.cs (hunk 3 above)
Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Web/Module.cs]: Fixed

File: {RepositoryName}/src/{Namespace}.Core/ModuleConstants.cs (hunk 5 above)
Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Core/ModuleConstants.cs]: Fixed

File: {RepositoryName}/src/{Namespace}.Web/package.json (hunk 4 above)
Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Web/package.json:1]: But in [OrdersModule](https://github.com/VirtoCommerce/vc-module-order/tree/release/3.0.0/src/VirtoCommerce.OrdersModule.Web) repository "package-lock.json" file is exist.
Need more comments on why "package-lock.json" file should be removed.

File: {RepositoryName}/src/{Namespace}.Web/module.manifest (hunk 6 above)
Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Web/module.manifest:21]: fixed

File: README.md [hunk 7]
```diff
@@ -28,12 +28,12 @@ Difference between this script and [Virto Commerce 3.x Module Templates](https:/
...
 
 ```command
-.\Module_Template.ps1 -ModuleName CustomerReviews -Author "Jon Doe" -CompanyName VirtoCommerce -PlatformVersion "3.0.0" -CoreModuleVersion "3.0.0-rc4-0225" -DataModuleVersion "3.0.0-rc4-0225"
+.\Module_Template.ps1 -ModuleName CustomerReviews -Author "Jon Doe" -CompanyName VirtoCommerce -PlatformVersion "3.0.0" -PlatformNuGetPackageVersion "3.0.0-rc4-0225"
 ```
 
 where `-ModuleName` your module name, `-Author` your name and `-CompanyName` your company name. This command will replace constants with their values in all files.
 
-6. `PlatformVersion` and `CoreModuleVersion` parameters have "3.0.0" "3.0.0-rc4-0225" default values respectively. You can skip this parameters if you use default Platform and CoreModule versions
+6. `PlatformVersion` and `PlatformNuGetPackageVersion` parameters have "3.0.0" "3.0.0-rc4-0225" default values respectively. You can skip this parameters if you use default versions
```
Comment (eMazeika) [inline README.md]: this -> these

Comment (mvktsk) [reply README.md]: fixed

File: {RepositoryName}/src/{Namespace}.Web/package.json (hunk 4 above)
Comment (eMazeika) [reply {RepositoryName}/src/{Namespace}.Web/package.json:1]: oh, an update: everyone SHOULD commit package-lock to source control.

File: {RepositoryName}/src/{Namespace}.Web/Scripts/module.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+                                isClosingDisabled: true
+                            };
+                            bladeNavigationService.showBlade(newBlade);
+                        }
+                    ]
+                });
+        }
+    ])
+    .run(['$rootScope', 'platformWebApp.mainMenuService', 'platformWebApp.widgetService', '$state',
+        function ($rootScope, mainMenuService, widgetService, $state) {
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Web/Scripts/module.js]: remove '$rootScope', it's never used in real modules

Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Web/Scripts/module.js]: fixed

Comment (eMazeika) [reply {RepositoryName}/src/{Namespace}.Web/Scripts/module.js]: and also remove $rootScope

File: {RepositoryName}/src/{Namespace}.Web/Module.cs [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+{
+    public class Module : IModule
+    {
+        public ManifestModuleInfo ModuleInfo { get; set; }
+
+        public void Initialize(IServiceCollection serviceCollection)
+        {
+            // database initialization
+            var configuration = serviceCollection.BuildServiceProvider().GetRequiredService<IConfiguration>();
+            var connectionString = configuration.GetConnectionString("VirtoCommerce.{ModuleId}.Web") ?? configuration.GetConnectionString("VirtoCommerce");
```
Comment (eMazeika) [inline {RepositoryName}/src/{Namespace}.Web/Module.cs]: remove ".Web

Comment (mvktsk) [reply {RepositoryName}/src/{Namespace}.Web/Module.cs]: fixed

**Review by eMazeika — APPROVED** [review]

**Pull request merged**